var/spack/repos/builtin/packages/py-dask/package.py

```diff
@@ -152,6 +157,7 @@ class PyDask(PythonPackage):
         "py-distributed@2020.12.0:2021.8.0", type=("build", "run"), when="@2020.12.0: +distributed"
```

Comment:
These older versions should have upper bounds in `when`, otherwise they will also apply to the latest version and will conflict with the requirements of newer versions. You should find that it isn't actually possible to concretize `py-dask@2022.10.2+distributed` because of this.

Reply:
They already have a trailing colon, but they shouldn't.

Reply:
If I remember, the `@a:b` is inclusive of `a` and `b`, so the `when` constraint should be `a:(b-1)` and `b:(c-1)`?

Reply:
Correct